test/index.test.ts

```diff
@@ -0,0 +1,145 @@
+import * as api from '../src/index';
+import WallyJS from '../src/wally-js';
+jest.mock('../src/wally-js');
+
+describe('Public API', () => {
+  let consoleSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    (WallyJS as jest.Mock).mockClear();
+    consoleSpy = jest.spyOn(console, 'error');
+    consoleSpy.mockImplementation(() => {});
+  });
+
+  afterEach(() => {
+    api.clearInstance();
+    consoleSpy.mockRestore();
+  });
+
+  describe('init', () => {
+    it('shows a console error when not called on the client', () => {
+      const windowSpy = jest.spyOn(window, 'window', 'get');
+      windowSpy.mockImplementation(undefined);
+
+      api.init({ clientId: '1234' });
+      expect(console.error).toHaveBeenCalled();
+
+      windowSpy.mockRestore();
+    });
+
+    it('creates a new wally instance', () => {
+      api.init({ clientId: '1234' });
+      expect(WallyJS).toHaveBeenCalled();
+    });
+
+    it('reuses existing wally instance', () => {
+      api.init({ clientId: '1234' });
+      api.init({ clientId: '1234' });
+      expect(WallyJS).toHaveBeenCalledTimes(1);
+      expect((WallyJS as jest.Mock).mock.instances.length).toBe(1);
+    });
+
+    it('checks for redirects', () => {
+      jest
+        .spyOn(WallyJS.prototype, 'isRedirected')
+        .mockImplementation(() => false);
+      api.init({ clientId: '1234' });
+      const mockWally = (WallyJS as jest.Mock).mock.instances[0];
+      expect(mockWally.isRedirected).toHaveBeenCalled();
+      expect(mockWally.handleRedirect).toHaveBeenCalledTimes(0);
+
+    });
+
+    it('handles redirects', () => {
+      jest
+        .spyOn(WallyJS.prototype, 'isRedirected')
+        .mockImplementation(() => true);
+      api.init({ clientId: '1234' });
+      const mockWally = (WallyJS as jest.Mock).mock.instances[0];
+      expect(mockWally.handleRedirect).toHaveBeenCalled();
+    });
+  });
+
+  describe('getProvider', () => {
+    it('returns undefined and shows error when no module instance', () => {
```

Comment:
Minor: Perhaps it may be better to return `null`. `null` represents an `empty` value whereas `undefined` is more of a possible error case that is not handled.